### ISSUE TITLE
[Suggestion] Ring Icon on Married Ships

### ISSUE THREAD

**Issue by ethrundr** [dragonjet/KC3Kai#501]
I know this is minor and seems unimportant, but as the actual game displays so, why won't we add a small ring icon next to the name (or level) of married ship(s)?

This can be done either in Strategy Room only or in the Panel as well :smiley:

**Comment by nicky008** [comment]
:+1: Having more details about my harems...

**Comment by ethrundr** [comment]
@nicky008 actually that's one of my hidden motives suggesting this :laughing:

**Comment by dragonjet** [comment]
XD

**Comment by McDohl** [comment]
and changed the text to certain pink like how kadokawa do it :smirk:

**Comment by LostLogia4** [comment]
How about a tab on strategy room to record the date of marriage?

**Comment by McDohl** [comment]
suddenly waifu thread :sweat_smile: but i approve for timestamp

**Comment by bamboo3250** [comment]
Yes, I definitely need a waifu tab after event :laughing: 

![screen shot 2015-08-07 at 6 22 12 pm](https://cloud.githubusercontent.com/assets/8078323/9133849/6e48517a-3d31-11e5-9217-8d4ed800b9f7.png)

**Comment by LostLogia4** [comment]
@bamboo3250 r u seriously plan to marry them all?
sidenote: Didn't have my waifu candidate here, but she's in my fleet to begin with, so...

**Comment by McDohl** [comment]
@bamboo3250 invite me to ur stream :+1:

**Comment by bamboo3250** [comment]
@LostLogia4 yes, I can and will do that whenever they reach Lv.99. Now my concern is who will be my first waifu, Kongou or Orel cruising team at once (all subs have the same exp now) :grin: 

@McDohl I'm new with the stream Q v Q

**Comment by LostLogia4** [comment]
@bamboo3250 I've already chosen mine. I'll be keeping it a secret for now though.

**Comment by bamboo3250** [comment]
@LostLogia4 invite me to your stream when you do wedding :relaxed:

**Comment by LostLogia4** [comment]
Seems like I have to open a Twitch account just for this...

**Comment by ethrundr** [comment]
for those who have harems (I included), having a tab to specify the marriage dates will be a very good idea :laughing: 

to be honest, I already forgot the exact date of my first two marriage lol XD
have 10 as per yesterday, when I marry Uzuki :laughing:

**Comment by bamboo3250** [comment]
@ethrundr I see, so you like flat-chest, don't you? :smirk:

**Comment by ethrundr** [comment]
@bamboo3250 loli, flat-chest, twintail, tsundere, silver hair
all these 5 is my favorites :smiley: 
and that's why Amatsukaze is very lethal to me......

**Comment by dragonjet** [comment]
whoa whoa whoa so you're a justice-bro nice

**Comment by nicky008** [comment]
Kamo too? :relaxed:

**Comment by ethrundr** [comment]
@nicky008 yep, kamo too :laughing: 
@dragonjet XD

**Comment by Crazyteitoku** [comment]
flat == loli! :laughing:

**Comment by ethrundr** [comment]
@Crazyteitoku Output: true XD

**Comment by ReiFan49** [comment]
testing things (with married flag is true lol)

![image](https://cloud.githubusercontent.com/assets/9847780/9156976/291e80e2-3f16-11e5-9586-5fa7dd0c48d0.png)
maybe it should have a ring icon... instead the placeholder ring thingy :>

(also, for the date, too lazy `_(:3/|_)_`)

**Comment by dragonjet** [comment]
hm oh wow, good for you guys.. I cannot test this. I have no married ship :cry:

**Comment by nicky008** [comment]
@ReiFan49 like this, the ring icon would be overlapped "Kai Ni" chars section for many ships here... `:v`

**Comment by ReiFan49** [comment]
sounds like i have some possibilities here:
- expand the table but only for ring icon beside the kanmusu name (like what i've done now, but wider table)

  (iirc 20-24px, don't worry, the ring is just a placeholder)
- put the ring icon besides the level, but the level starts from 1 to 51
  - do we need a zero-index or 1-index convention for this? a Lv.0 ring = Lv.100 or Lv.1 ring = Lv.100

**Comment by nicky008** [comment]
for the second choice, we should need 0-index.

**Comment by Yukinyaa** [comment]
Ring icon could be at bottom right of the kanmusu icon. That way it won't cover any strings .

**Comment by ReiFan49** [comment]
somehow it could be included as the another possibilities... i almost forgot that the idea passes my mind for a moment...

(preparing for 4 possible case)

**Comment by dragonjet** [comment]
I vote for case 1

**Comment by Yukinyaa** [comment]
Case 1 with smaller ring?

**Comment by ReiFan49** [comment]
1. Ring on kanmusu name
2. Ring on Level, colorize it, and 0-index convention (0-50)
  3. similar, but 1-index convention (1-51)
4. Ring on kanmusu icon (smallest ring)

**Comment by nicky008** [comment]
Case 1 with the ring's size as case 4...

**Comment by ReiFan49** [comment]
in that case, i'll set the default to be the first case,
(i don't know if the toggle to switch it should be the console tweaking based or direct toggle on settings lol, i'll leave it for now)

**Comment by ethrundr** [comment]
@ReiFan49 IMO the ring better to be placed at the very right end, after the heart-lock icon.
The shiplisting in-game also do it that way, actually.

**Comment by ReiFan49** [comment]
on that case table expansion is required by 2 item,
first one is ring icon, RIGHTmost one
second one is reinforcement expansion thingy, this one must be the fifth slot that devs had prepared all along faraway before this event.

**Comment by ethrundr** [comment]
to be honest, I dislikes the idea that it will be overlapping with the names or level (or any other details), so if possible, table expansion should be the best option

**Comment by ReiFan49** [comment]
![image](https://cloud.githubusercontent.com/assets/9847780/9217210/bb59d108-40ee-11e5-82ac-21cf8b5cd56f.png)
should i leave the So*ic ring? (i want to play maimai again :sob: )
or should i just remove it and wait for the other to include the true marriage ring?

(ps: table compression, to make it fit, with possible value input, which is cirno's)

**Comment by LostLogia4** [comment]
Is it possible to use this as the ring icon?
![183](https://cloud.githubusercontent.com/assets/12930396/9379919/36b6a2f6-475f-11e5-9352-1ceea681c13d.png)
(yes, extracted directly from game assets, jic there is a legal issue or something)

**Comment by ReiFan49** [comment]
well, most of the pictures stored on this extension had the similar/same one right?

**Comment by bamboo3250** [comment]
If possible, I would like to use the ring in the game (the image @LostLogia4 posted above). Our extenstion also uses images from the game assets so I think it would not be a big issue.

**Comment by nicky008** [comment]
Eh? But I thought we already had its picture. :anguished:

Alright...

**Comment by LostLogia4** [comment]
Hmm, I've managed to extract the individual frames of the animated version, but I couldn't find a way to combine it into a single gif...

**Comment by nicky008** [comment]
@LostLogia4 Don't need hurry up, just make sure these pictures are proper for using before upload here.

**Comment by ethrundr** [comment]
the static image @LostLogia4 has extracted above is enough imo....
but well, if we can manage to use gifs, why not :smiley:

**Comment by LostLogia4** [comment]
Here's the animated ring icon's frame, about 102 pic in total.
https://db.tt/OgKfTsam
As I've said, it has transparent background, so...

**Comment by bamboo3250** [comment]
Wow, 102 frames for a ring O.o

**Comment by ethrundr** [comment]
that 102 frames....
well it's not that surprising though, since the "shine" that being animated is making a whole circle around the ring....

**Comment by LostLogia4** [comment]
It's hell to extract it one by one by jpexs extractor.
Could've converted the commonassets to fla and then use adobe flash professional to extract it, thought it had disappointing results, turns out it's because of transparency messing with the gif creation.

**Comment by McDohl** [comment]
http://i.imgur.com/2OBVyue.gif

**Comment by ethrundr** [comment]
finally someone fuses the frames.... o.o
but isn't the shades from the light is bit too big?

**Comment by McDohl** [comment]
I don't know how that happen myself xD
I tried using photoshop this time but It's kinda meh http://i.imgur.com/q4mYKrK.gif

**Comment by LostLogia4** [comment]
Found [this one](https://jsfiddle.net/ryc913wu/7/) for the big ring. Haven't found a smaller one, but it seems like we gotta use CSS animation for this one.
